src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,465 @@
+
+    def add_group_range(self, group_start_mjd, group_end_mjd, frozen=True, j=None):
+        """Add an orbital piecewise parameter group range.
+        Parameters
+        ----------
+        group_start_mjd : np.float128
+                Lower group edge
+        group_end_mjd : np.float128
+                Upper group edge
+        j : int
+                Number to label the piece being added.
+        """
+        # check the validity of the desired group to add
+        if group_end_mjd is not None and group_start_mjd is not None:
+            if group_end_mjd <= group_start_mjd:
+                raise ValueError("Starting MJD is greater than ending MJD.")
+            elif j < 0:
+                raise ValueError(
+                    f"Invalid index for group: {j} should be greater than or equal to 0"
+                )
+            elif j > 9999:
+                raise ValueError(
+                    f"Invalid index for group. Cannot index beyond 9999 (yet?)"
+                )
+
+        i = f"{int(j):04d}"
+        self.add_param(
+            prefixParameter(
+                name="PLB_{0}".format(i),
+                units="MJD",
+                unit_template=lambda x: "MJD",
+                description="Beginning of paramX interval",
+                description_template=lambda x: "Beginning of paramX interval",
+                parameter_type="MJD",
+                time_scale="utc",
+                value=group_start_mjd,
+            )
+        )
+        self.add_param(
+            prefixParameter(
+                name="PUB_{0}".format(i),
+                units="MJD",
+                unit_template=lambda x: "MJD",
+                description="End of paramX interval",
+                description_template=lambda x: "End of paramX interval",
+                parameter_type="MJD",
+                time_scale="utc",
+                value=group_end_mjd,
+            )
+        )
+        self.setup()
+
+    def remove_range(self, index):
+        """Removes all orbital piecewise parameters associated with a given index/list of indices.
+        Parameters
+        ----------
+        index : float, int, list, np.ndarray
+            Number or list/array of numbers corresponding to T0X/A1X indices to be removed from model.
+        """
+        if (
+            isinstance(index, int)
+            or isinstance(index, float)
+            or isinstance(index, np.int64)
+        ):
+            indices = [index]
+        elif not isinstance(index, list) or not isinstance(index, np.ndarray):
+            raise TypeError(
+                f"index must be a float, int, list, or array - not {type(index)}"
+            )
+        for index in indices:
+            index_rf = f"{int(index):04d}"
+            for prefix in ["T0X_", "A1X_", "PLB_", "PUB_"]:
+                self.remove_param(prefix + index_rf)
+        self.validate()
+        self.setup()
+
+    def add_piecewise_param(self, param, j, param_unit=None, paramx=None):
```

Comment:
Since there is a small, defined list for this, could you have it work like:
`add_piecewise_params(j, **kwargs)`
and then parse kwargs for `T0=...` and `A1=...`?  And make sure they can accept floats or quantities (using the default units for the given parameters when only floats are passed)?

Reply:
Now operates as described